src/resources/js/dropdowns.js

```diff
@@ -14,307 +14,361 @@ var tribe_dropdowns = tribe_dropdowns || {};
+			args.regexSplit = new RegExp( args.regexSplitString, 'ig' );
+		}
 
-				// Allows HTML from Select2 AJAX calls
-				args.escapeMarkup = function (m) {
-					return m;
-				};
+		// Select also allows Tags, so we go with that too
+		if ( $select.is( '[data-tags]' ) ){
+			args.tags = $select.data( 'tags' );
 
-				args.ajax = { // instead of writing the function to execute the request we use Select2's convenient helper
-					dataType: 'json',
-					type: 'POST',
-					url: window.ajaxurl,
-					results: function ( data ) { // parse the results into the format expected by Select2.
-						return data.data;
-					}
-				};
+			args.initSelection = obj.init_selection;
+
+			args.createSearchChoice = function( term, data ) {
+				if ( term.match( args.regexToken ) ) {
+					return { id: term, text: term };
+				}
+			};
 
-				// By default only send the source
-				args.ajax.data = function( search, page ) {
-					return {
-						action: 'tribe_aggregator_dropdown_' + source,
-					};
+			if ( 0 === args.tags.length ){
+				args.formatNoMatches = function(){
+					return $select.attr( 'placeholder' );
 				};
 			}
+		}
 
-			$container = ( $select.select2( args ) ).select2( 'container' );
+		// When we have a source, we do an AJAX call
+		if ( $select.is( '[data-source]' ) ) {
+			var source = $select.data( 'source' );
 
-			if ( carryOverData.length > 0 ) {
-				carryOverData.map( function ( dataKey ) {
-					var attr = 'data-' + dataKey,
-						val = $select.attr( attr );
+			// For AJAX we reset the data
+			args.data = { results: [] };
 
-					if ( ! val ) {
-						return;
-					}
+			// Allows HTML from Select2 AJAX calls
+			args.escapeMarkup = obj.allow_html_markup;
 
-					this.attr( attr, val );
-				}, $container );
-			}
-		})
-		.on( 'select2-open', function( event ) {
-			var $select = $( this );
+			args.ajax = { // instead of writing the function to execute the request we use Select2's convenient helper
+				dataType: 'json',
+				type: 'POST',
+				url: window.ajaxurl,
+				results: function ( data ) { // parse the results into the format expected by Select2.
+					return data.data;
+				}
+			};
 
-			// If we have a placeholder for search, apply it!
-			if ( $select.is( '[data-search-placeholder]' ) ){
-				$( '.select2-input' ).attr( 'placeholder', $select.data( 'searchPlaceholder' ) );
-			}
-		} )
-		.on( 'select2-close', function( event ) {
-			var $select = $( this );
-
-			// Regardless if it had any search-placeholder we remove it
-			$( '.select2-input' ).attr( 'placeholder', null );
-		} )
-		.on( 'change', function( event ) {
-			var $select = $(this),
-				data = $( this ).data( 'value' );
-
-			if ( ! $select.is( '[multiple]' ) ){
-				return;
-			}
-			if ( ! $select.is( '[data-source]' ) ){
-				return;
-			}
+			// By default only send the source
+			args.ajax.data = function( search, page ) {
+				return {
+					action: 'tribe_aggregator_dropdown_' + source,
```

Comment:
This (piece of JS as a whole, if not this specific part of it) is already used by more than just aggregator - which makes sense as it's in tribe-common. Isn't it a recipe for trouble if in future we want other components to listen out for an act on this?